app/src/androidTest/java/org/piwigo/ui/launcher/LoginTest.java

```diff
@@ -0,0 +1,126 @@
+
+        editUser.perform(replaceText("seessmall"), closeSoftKeyboard());
+        editUser.perform(pressImeActionButton());
+
+        ViewInteraction editPassword = onView(withId(R.id.password));
+        editPassword.perform(replaceText("seessmall"), closeSoftKeyboard());
+        editPassword.perform(pressImeActionButton());
+
+        ViewInteraction loginButton = onView(allOf(withId(R.id.login_button), withContentDescription("Login")));
+        loginButton.perform(scrollTo(), click());
+
+        // Added a sleep statement to match the app's execution delay.
+        // The recommended way to handle such scenarios is to use Espresso idling resources:
+        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction toolbar = onView(allOf(withText("Albums"), isDisplayed()));
+        toolbar.check(matches(isDisplayed()));
+
+        onView(withText("Small")).check(matches(isDisplayed()));
+        onView(withText("Large")).check(doesNotExist());
+
+        onView(withId(R.id.drawer_layout)).perform(open());
+
+        ViewInteraction navigation = onView(allOf(withText("Manage Accounts"), isDisplayed()));
+        navigation.perform(click());
+
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction addAccount = onView(allOf(withId(R.id.action_add_account), isDisplayed()));
+        addAccount.perform(click());
+
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        editUser = onView(withId(R.id.username));
+        editUser.check(matches(withText("")));
+
+        editURL = onView(withId(R.id.url));
+        editURL.perform(replaceText("https://tg1.kulow.org"));
+
+        editUser.perform(replaceText("seeslarge"));
+        editUser.perform(pressImeActionButton());
+
+        editPassword = onView(withId(R.id.password));
+        editPassword.perform(replaceText("seeslarge"));
+        editPassword.perform(pressImeActionButton());
+
+        loginButton = onView(allOf(withId(R.id.login_button), withContentDescription("Login")));
+        loginButton.perform(scrollTo(), click());
+
+        // Added a sleep statement to match the app's execution delay.
+        // The recommended way to handle such scenarios is to use Espresso idling resources:
+        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        // add account stays in 'manage account' - I think it should jump to it?
```

Comment:
Login to the first account jumps to the albums view while adding a 2nd account stays in 'manage accounts'. I kind of expect that the login mask of the 2nd account jumps to the album view as well. Can I change that?

Reply:
This is a good question. I also stumbled once over this inconsistency. Also the way we currently handle the selection of another account by directly activating and closing the AccountManager is a bit weird as it is impossible for example to delete an account that is currently not the active one, while on the other hand the focus should clearly be on simplification (and reducing number of clicks to) switch between accounts, so that one might be good to be fast. Also your proposal to directly close the AccountManager after creating a new account is consistent with that...

So yes, give it a try I think it makes sense (for me without being a UI expert)